sdk/data/azcosmos/cosmos_container_read_many.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+import (
+	"context"
+	"errors"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos/queryengine"
+)
+
+// Executes a query using the provided query engine.
+func (c *ContainerClient) executeReadManyWithEngine(queryEngine queryengine.QueryEngine, items []ItemIdentity, readManyOptions *ReadManyOptions, operationContext pipelineRequestOptions) ([]ReadManyItemsResponse, error) {
+	// NOTE: The current interface for runtime.Pager means we're probably going to risk leaking the pipeline, if it's provided by a native query engine.
+	// There's no "Close" method, which means we can't call `queryengine.QueryPipeline.Close()` when we're done.
+	// We _do_ close the pipeline if the user iterates the entire pager, but if they don't we don't have a way to clean up.
+	// To mitigate that, we expect the queryengine.QueryPipeline to handle setting up a Go finalizer to clean up any native resources it holds.
+	// Finalizers aren't deterministic though, so we should consider making the pager "closable" in the future, so we have a clear signal to free the native resources.
+
+	// if empty list of items, return empty list
+	if len(items) == 0 {
```

Comment:
We could probably promote this to the `ReadMany` API itself, just in case we do have alternate implementations of ReadMany. No matter which one we use, and empty list of items should always yield an empty result set.